lib/rule-list.js

```diff
@@ -155,17 +155,181 @@ export const SIMPLIFY_FRACTION = defineRuleString(
+    ? node.args[0].args[0]
+    : node.args[0]
+
+export const getDenominator = (node) => query.isNeg(node)
+    ? node.args[0].args[1]
+    : node.args[1]
+
+export const isDecimal = (node) => query.isNumber(node) && query.getValue(node) % 1 != 0
+
+export const decimal_to_fraction = (node) => {
+    // split up the decimal
+    const [int, dec] = node.value.toString().split('.')
+
+    // e.g. .2 -> 2/10
+    const fraction = build.div(build.number(parseInt(dec)), build.number(parseInt('1' + '0'.repeat(dec.length))))
+
+    // simplified the fraction if possible 
+    const simplified = canApplyRule(SIMPLIFY_FRACTION, fraction)
+          ? applyRule(SIMPLIFY_FRACTION, fraction)
+          : fraction
+
+    const newNumerator = parseInt(int) * query.getValue(simplified.args[1]) + query.getValue(simplified.args[0])
+    return build.div(build.number(newNumerator), simplified.args[1])
+}
+
+// e.g. 2 + 3/2 -> 2/1 + 3/2
+// e.g. 1.2 + 3/2 -> 6/5 + 3/2
+export const CONVERT_TO_FRACTION = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            // only some of the terms are fractions
+            const hasFraction = terms.some(term => isFraction(term)) && !terms.every(term => isFraction(term))
+
+            return hasFraction ? {node} : null
+        }
+    },
+
+    (node) => {
+        let terms = node.args
+
+        const negatives = getNegatives(terms)
+
+        // transform constant terms
+        terms = terms.map(term => isDecimal(term) && query.isNeg(term)
+                          ? decimal_to_fraction(term.args[0])
+                          : isDecimal(term)
+                          ? decimal_to_fraction(term)
+                          : isPolynomialTerm(term) && query.isNeg(term)
+                          ? build.div(term.args[0], build.number(1))
+                          : isPolynomialTerm(term)
+                          ? build.div(term, build.number(1))
+                          : query.isNeg(term)
+                          ? term.args[0]
+                          : term)
+
+        const result = build.add(
+            ...terms.map((term, i) => negatives[i]
+                         ? build.neg(term, {wasMinus: true})
+                         : term))
+        return result
+    }
+)
+
+// Finds common denominator in any scenario
+
+// e.g. 2/6 + 1/4 -> (2 * 2) / (6 * 2) + (1 * 3) / (4 * 3)
+// e.g. 2/(3 + x) + 2/3 -> (2 * 3)/ (3 * (3 + x)) + 2 * (3 + x) / (3 * (3 + x))
+// e.g. 3 + 2/3 -> (3 * 3) / 3 + 2 / 3
+export const COMMON_DENOMINATOR = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            
+            // checks if at least one of the terms is a fraction
+            const hasFraction = terms.some(term => isFraction(term))
+ 
+            let sameDenom
+            // rule cannot apply if all of the terms are fractions
+            // and they all have the same denominator
```

Comment:
It might be worth moth this comment up to the top of the function so that people don't have to read the whole function to understand the restrictions.